libcxx/benchmarks/algorithms/mismatch.bench.cpp

```diff
@@ -0,0 +1,31 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <algorithm>
+#include <benchmark/benchmark.h>
+#include <random>
+
+template <class T>
+static void bm_mismatch(benchmark::State& state) {
```

Comment:
I don't know how much you care for it but you can try to mess with data alignment.

ve1.data() and vec2.data() will be aligned to 16 bytes. Which can lead to loads being aligned.

the difference can be quite huge.

There are some things you can do about that, I don't know if they are worth it for 2 range algorithms.

At the very least maybe aligned your .data() pointers to 64 bytes:

* allocate both vectors with +64 bytes.
* align the pointers forward - get a span.

Reply:
I've added a TODO for now. Though even if it makes a difference I'm not sure we can do much about it, can we?

Reply:
You can align onr of the arrays. Also this makes for a better comparison